### PR TITLE
Add better support for mysql in `transition_conflict_error?` method

### PR DESCRIPTION
**Pull request by greysteil**
Previously ~~the MySQL tests~~ all the transition conflict error tests were passing on Rails 6 because the index name included the bits we expecting.

In the wild, people don't always have nice names for their indexes, so relying on them including the names of the columns we're interested in isn't robust. This PR looks up the indexes properly.

### CODE REVIEW

**Comment by greysteil** [comment]
@lawrencejones / @hmac can I get a review?

Should save you some pain when you upgrade to Rails 6. :octocat:

File: lib/statesman/adapters/active_record.rb [hunk 1]
```diff
@@ -128,10 +128,27 @@ def serialized?(transition_class)
...
+          indexes(transition_class.table_name).
+          select do |index|
+            next unless index.unique
+
+            index.columns.sort == [parent_join_foreign_key, "sort_key"] ||
+              index.columns.sort == [parent_join_foreign_key, "most_recent"]
+          end
+      end
+
+      def parent_join_foreign_key
```
Comment (lawrencejones) [inline lib/statesman/adapters/active_record.rb]: Do we have any guarantee that people will name their columns like this? Seems like a more solid assumption than the one you've removed in this PR, but was wondering if we should do something smarter for this too?

Comment (greysteil) [reply lib/statesman/adapters/active_record.rb]: Good shout. Found a smarter way to do it.

**Comment by greysteil** [comment]
@lawrencejones updated and working for all supported Rails versions. Can you take a second look?

**Comment by lawrencejones** [comment]
I'm good with this but am away from a keyboard. I'll either wait for Harry to pick it up or handle it tomorrow morning when I'm back in work. Just want to try it out locally to double check everything before pressing merge!

**Comment by greysteil** [comment]
Awesome, thanks @lawrencejones!

**Comment by lawrencejones** [comment]
Minor bump sound sensible to you, @greysteil?

**Comment by greysteil** [comment]
Yes please!

**Comment by lawrencejones** [comment]
Should be released now: https://github.com/gocardless/statesman/releases/tag/v4.1.0. No doubt dependabot is busy opening PRs as we speak!

**Pull request merged**